Examples/Algorithms/TrackFinding/src/HoughTransformSeeder.cpp

```diff
@@ -0,0 +1,529 @@
+    if (meas->layer != layer)
+      continue;
+    if (!((*(m_cfg.inSliceMeasurement))(meas->z, meas->layer, subregion)))
+      continue;
+
+    // This scans over y (pT) because that is more efficient in memory, in C.
+    // Unknown if firmware will want to scan over x instead.
+    for (unsigned y_ = 0; y_ < m_cfg.m_imageSize_y; y_++) {
+      unsigned y_bin_min = y_;
+      unsigned y_bin_max = (y_ + 1);
+
+      // Find the min/max x bins
+      auto xBins =
+          yToXBins(y_bin_min, y_bin_max, meas->radius, meas->phi, meas->layer);
+      // Update the image
+      for (unsigned y = y_bin_min; y < y_bin_max; y++)
+        for (unsigned x = xBins.first; x < xBins.second; x++) {
+          image(y, x).first++;
+          image(y, x).second.insert(
+              index +
+              spacepoints
+                  .size());  // add spacepoints size to differentiate them
+        }
+    }
+  }
+
+  return image;
+}
+
+ActsExamples::Image ActsExamples::HoughTransformSeeder::createImage(
+    std::vector<const SimSpacePoint*>& spacepoints,
+    std::vector<std::shared_ptr<const MeasurementStruct>>& meas,
+    int subregion) const {
+  ActsExamples::Image image(m_cfg.m_imageSize_y, m_cfg.m_imageSize_x);
+
+  for (unsigned i = 0; i < m_cfg.m_nLayers; i++) {
+    Image layerImage = createLayerImage(i, spacepoints, meas, subregion);
+    for (unsigned x = 0; x < m_cfg.m_imageSize_x; ++x)
+      for (unsigned y = 0; y < m_cfg.m_imageSize_y; ++y)
+        if (layerImage(y, x).first > 0) {
+          image(y, x).first++;
+          image(y, x).second.insert(layerImage(y, x).second.begin(),
+                                    layerImage(y, x).second.end());
+        }
+  }
+
+  return image;
+}
+
+bool ActsExamples::HoughTransformSeeder::passThreshold(Image const& image,
+                                                       unsigned x,
+                                                       unsigned y) const {
+  // Pass window threshold
+  unsigned width = m_cfg.m_threshold.size() / 2;
+  if (x < width || (image.size(1) - x) < width)
+    return false;
+  for (unsigned i = 0; i < m_cfg.m_threshold.size(); i++) {
+    if (image(y, x - width + i).first < m_cfg.m_threshold[i])
+      return false;
+  }
+
+  // Pass local-maximum check, if used
+  if (m_cfg.m_localMaxWindowSize)
+    for (int j = -m_cfg.m_localMaxWindowSize; j <= m_cfg.m_localMaxWindowSize;
+         j++)
+      for (int i = -m_cfg.m_localMaxWindowSize; i <= m_cfg.m_localMaxWindowSize;
+           i++) {
+        if (i == 0 && j == 0)
+          continue;
+        if (y + j < image.size(0) && x + i < image.size(1)) {
+          if (image(y + j, x + i).first > image(y, x).first)
+            return false;
+          if (image(y + j, x + i).first == image(y, x).first) {
+            if (image(y + j, x + i).second.size() > image(y, x).second.size())
+              return false;
+            if (image(y + j, x + i).second.size() ==
+                    image(y, x).second.size() &&
```

Comment:
having y first and x second as parameters for image gets my heart rate up, I didn't check if it's correct, if so it'd confuse me even more 😅